executor/show.go

```diff
@@ -388,7 +388,9 @@ func (e *ShowExec) fetchShowCreateTable() error {
 			if !mysql.HasNoDefaultValueFlag(col.Flag) {
 				switch col.DefaultValue {
 				case nil:
-					buf.WriteString(" DEFAULT NULL")
+					if !mysql.HasPriKeyFlag(col.Flag) {
```

Comment:
primary key column can't be NULL...
if a column has `mysql.HasPriKeyFlag`, it must have `mysql.HasNotNullFlag`